torch/_dynamo/compiled_autograd.py

```diff
@@ -308,6 +308,31 @@ def move_graph_nodes_to_cuda(self, graph) -> List[int]:
 
         return []
 
+    def mark_dims_unbacked(self):
+        # avoid zero-one specialization
+        # returns unbacked_inputs: input idx -> dim indices
+        unbacked_inputs = {}
+        inputs_node = next(iter(self.fx_tracer.graph.nodes))
+        for node in self.fx_tracer.graph.find_nodes(
+            op="call_function", target=operator.getitem
+        ):
+            if len(node.args) != 2 or node.args[0] != inputs_node:
+                continue
+            assert isinstance(node.args[1], int)
+            assert "tensor_meta" in node.meta
+            unbacked_sizes = []
+            for i,size in enumerate(node.meta["tensor_meta"].shape):
+                if isinstance(size, torch.SymInt) and size.node.hint < 2:
```

Comment:
Wonder why we check `size_node.hint < 2`, if it is < 2, does that mean it has already compiled a version for 0 or 1? Even if we mark it as "unbacked", we can not avoid the existing compilation for 0 or 1

Reply:
for sizes 0 and 1, by default we compile a graph that's only valid for that size (zero-one specialization). marking it unbacked hides the actual value of the size from the compiler, and the compiler is forced into compiling something that works for all shapes.

Reply:
I see, so if `size.node.hint >= 2`, then we don't need to mark it unbacked, is that right?